firmware/controllers/core/state_sequence.h

```diff
@@ -35,42 +37,65 @@
  */
 typedef trigger_value_e pin_state_t;
 
-/**
- * This class represents one channel of a digital signal state sequence
- * Each element represents either a HIGH or LOW state - while at the moment this
- * is not implemented using a bit array, it could absolutely be a bit array
- *
- * This sequence does not know anything about signal lengths - only signal state at a given index
- * This sequence can have consecutive zeros and ones since these sequences work as a group within MultiChannelStateSequence
- *
- * @brief   PWM configuration for the specific output pin
- */
-class SingleChannelStateSequence {
+template<typename base_t, unsigned n_elem, typename... tail_t>
```

Comment:
please note how you remove 6 lines of code comments while adding how many?

we need some basic code comments.